cql3/restrictions/single_column_primary_key_restrictions.hh

```diff
@@ -187,6 +187,7 @@ public:
                             join(", ", get_column_defs())));
         }
         do_merge_with(::static_pointer_cast<single_column_restriction>(restriction));
+        this->expression = make_conjunction(this->expression, restriction->expression);
```

Comment:
std::move(), at least for this->expression, and please check if we can move the rhs too.

Reply:
> std::move(), at least for this->expression,

Why?  Does it still apply after shared_ptr is eliminated?

> and please check if we can move the rhs too.

I argued several times that we cannot, since that invalidates the restriction parameter, and I don't want that risk now, because merge_with will go away anyway in the next step of my plan.

Reply:
Copying an expression is more expensive than copying a shared_ptr (it's a deep copy). I can live without the rhs being moved for a while, but there's no reason not to move the lhs.